### ISSUE TITLE
Deprecate old nwb_reader classes, functionality and tests

### ISSUE THREAD

**Issue by wbwakeman** [AllenInstitute/ipfx#355]
The nwb_reader work has made obsolete a number of classes, functionality and tests.
In order to keep it clean and easy to understand for users and our future selves, remove these

**Comment by NileGraddis** [comment]
my current understanding:

The new public interface is `ipfx.dataset.ephys_data_set.EphysDataSet`. This interface replaces:
- `ipfx.ephys_data_set.EphysDataSet`
- `ipfx.aibs_data_set.AibsDataSet`
- `ipfx.hbg_data_set.HbgDataSet`
- `ipfx.mies_data_set.MiesDataSet`

It does task by relying on injected dependencies, which extract information from NWB2 files of various origin and provide a stable interface defined by `ipfx.dataset.ephys_data_interface.EphysDataInterface`.

The specific public methods implemented on `EphysDataSet` are (comments mine):
```
extract_sweep_info(self) -> Dict[str,Any] # where is the spec for sweep_info?
build_sweep_table # creates a public dataframe attribute, sweep_table, but why is this method public? 
filtered_sweep_table(self, clamp_mode=None, stimuli=None, stimuli_exclude=None) # -> pd.DataFrame
get_sweep_number(self, stimuli, clamp_mode=None) # -> int
voltage_current(self, sweep_data, clamp_mode) -> Tuple[np.arrray,np.array]
sweep(self, sweep_number) -> Sweep
sweep_set(self, sweep_numbers) -> SweepSet

also a staticmethod: # why is this a staticmethod?
drop_trailing_zeros_in_response(sweep_data)
```

The replaced classes have several methods not implemented on this abc:
```
EphysDataSet # the old one
-------------------
nwb_data(self): # correctly excluded!
add_clamp_mode(self, sweep_info) # not sure why this is public, or an instance method
get_sweep_record(self, sweep_number) # a convenience for getting a row of the sweep table as dict
aligned_sweeps(self, sweep_numbers, stim_onset_delta)
extract_sweep_stim_info(self)
get_recording_date(self)
get_sweep_data(self, sweep_number)
get_clamp_mode(self,stimulus_number)
get_stimulus_code(self,stimulus_number)
get_stimulus_name(self, stim_code)

AibsDataSet(EphysDataSet)
--------------------------------------
get_stimulus_code_ext(self, stim_code,sweep_num)
get_stimulus_units(self, sweep_num)
# it looks like these 2 are implicitly part of the old interface

HbgDataSet(EphysDataSet)
-----------------------------------
get_stimulus_units(self, sweep_num)
get_stimulus_code_ext(self, sweep_num)
extract_sweep_record

MiesDataSet(HbgDataSet)
-----------------------------------
# everything covered by ancestors

```
In summary, we do not have specified a replacement for the old `EphysDataSet` (and that's without getting into implicitly defined data!). We talked about starting with just the components required for `run_pipeline`, so it might be that all of these missing fields are simply unnecessary for `run_pipeline`.

versioning
--------------
Ultimately, our #286-derived work will result in a major version increment:
- (definitely) since we are relegating NWB1 support to a tag
- (possibly) if we change the `EphysDataSet` interface, as shown above

We should probably deprecate first, but the scope of our deprecation depends on the details of our changes to the `EphysDataSet` interface.

**Comment by NileGraddis** [comment]
here's what I think needs to be done here:
- [ ] make a pr against master (or dev, not completely sure what the branching strat is).
    - [ ] should deprecate construction outside of `create_data_set`
    - [ ] should deprecate a couple of internal-only-seeming methods and attributes not present in the new version (`build_sweep_table`, `nwb_data`, `add_clamp_mode`).
    - [ ] should deprecate NWB1-dependent code
    - [ ] should mention in changes / readme that py2 support and nwb1 support are being dropped
- make a pr against 286 dev branch that
    - [ ] removes all the old  dataset code and tests
    - [ ] updates all constructors of datasets to use the new dependency-injection interface
    - [ ] caches some NWB2 example data

**Comment by NileGraddis** [comment]
#372 and #373 now cover this work, closing
